src/emulator/remoteconfig/index.ts

```diff
@@ -0,0 +1,301 @@
+  files: SourceFile[];
+}
+
+export interface SourceFile {
+  name: string;
+  content: string;
+}
+
+export interface ValidTemplate {
+  valid: boolean;
+  msg: string;
+}
+
+export class RemoteConfigEmulator implements EmulatorInstance {
+  private destroyServer?: () => Promise<void>;
+  private _app?: express.Express;
+  private _templateWatcher?: chokidar.FSWatcher;
+  private _templateSource?: Source;
+  private _emulatorTemplate?: any;
+  private _cloudFunctions: RemoteConfigCloudFunctions;
+
+  private _logger = EmulatorLogger.forEmulator(Emulators.REMOTE_CONFIG);
+
+  constructor(private args: RemoteConfigEmulatorArgs) {
+    this._cloudFunctions = new RemoteConfigCloudFunctions(args.projectId);
+  }
+
+  get logger(): EmulatorLogger {
+    return this._logger;
+  }
+
+  get template(): any {
+    return this._emulatorTemplate;
+  }
+
+  set template(newTemplate: any) {
+    this._emulatorTemplate = newTemplate;
+    this._cloudFunctions.dispatch("update", {
+      description: "Emulator template updated.",
+      updateOrigin: "REMOTE_CONFIG_UPDATE_ORIGIN_UNSPECIFIED",
+      updateTime: new Date().toISOString(),
+      updateType: "REMOTE_CONFIG_UPDATE_TYPE_UNSPECIFIED",
+      updateUser: {
+        email: "emulator@example.com",
+      },
+      versionNumber: 1,
+    });
+  }
+
+  async start(): Promise<void> {
+    const { host, port } = this.getInfo();
+    this._logger.logLabeled("BULLET", "remote config", `Emulator loading...`);
+
+    this._app = await createApp(this.args.projectId, this);
+
+    if (typeof this.args.template === "string") {
+      const templateFile = this.args.template;
+      this.updateTemplateSource(templateFile);
+    } else {
+      this._templateSource = this.args.template;
+    }
+
+    if (!this._templateSource || this._templateSource.files.length === 0) {
+      throw new FirebaseError(
+        "Can not initialize Remote Config emulator without a template source / file."
+      );
+    } else if (this._templateSource.files.length > 1) {
+      throw new FirebaseError(
+        "Can not initialize Remote Config emulator with more than one template source / file."
+      );
+    }
+
+    this.loadTemplate(this._templateSource);
+
+    const templatePath = this._templateSource.files[0].name;
+    this._templateWatcher = chokidar.watch(templatePath, { persistent: true, ignoreInitial: true });
+    this._templateWatcher.on("change", async () => {
+      // There have been some race conditions reported (on Windows) where reading the
+      // file too quickly after the watcher fires results in an empty file being read.
+      // Adding a small delay prevents that at very little cost.
+      await new Promise((res) => setTimeout(res, 5));
+
+      this._logger.logLabeled(
+        "BULLET",
+        "remoteconfig",
+        `Change detected, updating template for Remote Config...`
+      );
+      this.updateTemplateSource(templatePath);
+      this.loadTemplate(this._templateSource);
```

Comment:
Might want to use debounce here (https://github.com/firebase/firebase-tools/blob/master/src/utils.ts#L710) to prevent this from being called excessively when chokidar fires multiple times in quick succession.

We do something very similar when watching for changes to functions files (https://github.com/firebase/firebase-tools/blob/27957b30c5ed67e5fdf05a6dea3240eb2f9f7ef1/src/emulator/functionsEmulator.ts#L447), and it generally improves that feel of the watch experience.